ecs-agent/ecs/model/api/docs-2.json

```diff
@@ -0,0 +1,4197 @@
+      }
+    },
+    "ContainerInstances": {
+      "base": null,
+      "refs": {
+        "DescribeContainerInstancesResponse$containerInstances": "<p>The list of container instances.</p>",
+        "UpdateContainerInstancesStateResponse$containerInstances": "<p>The list of container instances.</p>"
+      }
+    },
+    "ContainerOverride": {
+      "base": "<p>The overrides that are sent to a container. An empty container override can be passed in. An example of an empty container override is <code>{\"containerOverrides\": [ ] }</code>. If a non-empty container override is specified, the <code>name</code> parameter must be included.</p> <p>You can use Secrets Manager or Amazon Web Services Systems Manager Parameter Store to store the sensitive data. For more information, see <a href=\"https://docs.aws.amazon.com/AmazonECS/latest/developerguide/secrets-envvar.html\">Retrieve secrets through environment variables</a> in the Amazon ECS Developer Guide.</p>",
+      "refs": {
+        "ContainerOverrides$member": null
+      }
+    },
+    "ContainerOverrides": {
+      "base": null,
+      "refs": {
+        "TaskOverride$containerOverrides": "<p>One or more container overrides that are sent to a task.</p>"
+      }
+    },
+    "ContainerStateChange": {
+      "base": "<p>An object that represents a change in state for a container.</p>",
+      "refs": {
+        "ContainerStateChanges$member": null
+      }
+    },
+    "ContainerStateChanges": {
+      "base": null,
+      "refs": {
+        "SubmitTaskStateChangeRequest$containers": "<p>Any containers that's associated with the state change request.</p>"
+      }
+    },
+    "Containers": {
+      "base": null,
+      "refs": {
+        "Task$containers": "<p>The containers that's associated with the task.</p>"
+      }
+    },
+    "CreateCapacityProviderRequest": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateCapacityProviderResponse": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateClusterRequest": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateClusterResponse": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateServiceRequest": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateServiceResponse": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateTaskSetRequest": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CreateTaskSetResponse": {
+      "base": null,
+      "refs": {
+      }
+    },
+    "CredentialProviderInvocationException": {
```

Comment:
Same question as [this comment](https://github.com/aws/amazon-ecs-agent/pull/3974#discussion_r1370647495) but for `CredentialProviderInvocationException`.

Reply:
CredentialProviderInvocationException had the same issue as the other shapes mentioned [here](https://github.com/aws/amazon-ecs-agent/pull/3974#discussion_r1370749152). CredentialProviderInvocationException was removed for being internal only  shape. The operations which were referring to this shape were: StartTask and RunTask. Instead of removing the operations as a whole, just the error referring to the CredentialProviderInvocationException shape was removed. I will discuss with you offline if removing the whole operations is the right way to go.

Reply:
I will remove the corresponding operation or shape in the docs as well after our discussion.